app/src/main/java/emu/skyline/KeyReader.kt

```diff
@@ -26,60 +26,53 @@ object KeyReader {
      * Reads keys file, trims and writes to internal app data storage, it makes sure file is properly formatted
      */
     fun import(context : Context, uri : Uri, keyType : KeyType) : Boolean {
-        Log.i(Tag, "Parsing ${keyType.name}")
+        Log.i(Tag, "Parsing ${keyType.name} $uri")
 
         if (!DocumentFile.isDocumentUri(context, uri))
             return false
 
-        val fileName = DocumentFile.fromSingleUri(context, uri)!!.name
-        if (fileName?.substringAfterLast('.')?.startsWith("keys")?.not() ?: false)
-            return false
-
         val tmpOutputFile = File("${context.filesDir.canonicalFile}/${keyType.fileName}.tmp")
 
         val inputStream = context.contentResolver.openInputStream(uri)
-        val outputStream = tmpOutputFile.bufferedWriter()
-
-        val valid = inputStream!!.bufferedReader().useLines {
-            for (line in it) {
-                val pair = line.split("=")
-                if (pair.size != 2)
-                    return@useLines false
-
-                val key = pair[0].trim()
-                val value = pair[1].trim()
-                when (keyType) {
-                    KeyType.Title -> {
-                        if (key.length != 32 && !isHexString(key))
-                            return@useLines false
-                        if (value.length != 32 && !isHexString(value))
-                            return@useLines false
-                    }
-                    KeyType.Prod -> {
-                        if (!key.contains("_"))
-                            return@useLines false
-                        if (!isHexString(value))
-                            return@useLines false
+        tmpOutputFile.bufferedWriter().use { writer ->
+            val valid = inputStream!!.bufferedReader().useLines {

```

Comment:
Is the not-null assertion operator avoidable here? I'd avoid it in all cases when possible, maybe by adding some error handling and displaying a Toast/Snackbar to the user if necessary.

Reply:
`inputStream` can only be `null` when the content provider has crashed. This shouldn't happen, unless there is something wrong with the OS